tests/test_nock_lifecycle.js

```diff
@@ -1,215 +1,213 @@
 
-  expect(scope.isDone()).to.be.true()
+      expect(scope.isDone()).to.be.false()
 
-  expect(onResponse).to.have.been.calledOnce()
-})
+      nock.activate()
+      expect(nock.isActive()).to.be.true()
 
-test('clean all works', async t => {
-  nock('http://example.test')
-    .get('/')
-    .twice()
-    .reply()
+      expect(await got(url)).to.include({ statusCode: 304 })
 
-  await got('http://example.test/')
+      expect(scope.isDone()).to.be.true()
 
-  nock.cleanAll()
+      expect(onResponse).to.have.been.calledOnce()
 
-  await assertRejects(got('http://example.test/'), err => {
-    expect(err).to.include({ name: 'RequestError', code: 'ENOTFOUND' })
-    return true
+      server.close()
+    })
   })
-})
 
-test('cleanAll should remove pending mocks from all scopes', t => {
-  const scope1 = nock('http://example.test')
-    .get('/somepath')
-    .reply(200, 'hey')
-  expect(scope1.pendingMocks()).to.deep.equal([
-    'GET http://example.test:80/somepath',
-  ])
-  const scope2 = nock('http://example.test')
-    .get('/somepath')
-    .reply(200, 'hey')
-  expect(scope2.pendingMocks()).to.deep.equal([
-    'GET http://example.test:80/somepath',
-  ])
-
-  nock.cleanAll()
-
-  expect(scope1.pendingMocks()).to.be.empty()
-  expect(scope2.pendingMocks()).to.be.empty()
-  t.end()
-})
+  describe('`cleanAll()`', () => {
+    it('removes a half-consumed mock', async () => {
+      nock('http://example.test')
+        .get('/')
+        .twice()
+        .reply()
 
-test('cleanAll removes persistent mocks', async t => {
-  nock('http://example.test')
-    .persist()
-    .get('/')
-    .reply()
+      await got('http://example.test/')
 
-  nock.cleanAll()
+      nock.cleanAll()
 
-  await assertRejects(got('http://example.test/'), err => {
-    expect(err).to.include({
-      name: 'RequestError',
-      code: 'ENOTFOUND',
+      await assertRejects(got('http://example.test/'), err => {
+        expect(err).to.include({ name: 'RequestError', code: 'ENOTFOUND' })
+        return true
+      })
     })
-    return true
-  })
-})
-
-test('is done works', async t => {
-  nock('http://example.test')
-    .get('/')
-    .reply(200)
 
-  expect(nock.isDone()).to.be.false()
-
-  await got('http://example.test/')
-
-  expect(nock.isDone()).to.be.true()
-})
+    it('removes pending mocks from all scopes', () => {
+      const scope1 = nock('http://example.test')
+        .get('/somepath')
+        .reply(200, 'hey')
+      expect(scope1.pendingMocks()).to.deep.equal([
+        'GET http://example.test:80/somepath',
+      ])
+      const scope2 = nock('http://example.test')
+        .get('/somepath')
+        .reply(200, 'hey')
+      expect(scope2.pendingMocks()).to.deep.equal([
+        'GET http://example.test:80/somepath',
+      ])
+
+      nock.cleanAll()
+
+      expect(scope1.pendingMocks()).to.be.empty()
+      expect(scope2.pendingMocks()).to.be.empty()
+    })
 
-test('isDone', async t => {
```

Comment:
Probably easier to see in the side-by-side diff that these two tests of `isDone()` are doing the same thing.